client/extensions/hello-dolly/state/selectors.js

```diff
@@ -0,0 +1,19 @@
+
+/**
+ * External Dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal Dependencies
+ */
+import lyrics from './lyrics';
+
+export const getCurrentLyric = ( lines ) => ( reduxState ) => {
```

Comment:
The ambiguity here is whether or not we're passing the entire redux state tree or a sub-node of that state that applies to this reducer. For instance, the reducer gets passed the sub-node, but this selector expects the top-level redux state tree.

I do feel we'll have to figure this out at a much larger level at some point. Personally, I'm not a big fan of passing around the entire redux state tree all the time, but that's a much larger discussion.